### PR TITLE
Bug 1808973: [WIP] azurerm terraform fixes for address_prefixes

### PR DESCRIPTION
**Pull request by jhixson74**
When a subnet uses address_prefixes, address_prefix is null in the virtual network resource. Add address_prefixes to virtual network resource and allow null for address_prefix. Don't assume IPv4 for private_ip_address on load balancer resource. 

https://bugzilla.redhat.com/show_bug.cgi?id=1808973

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
/hold

We are modifying the vendor here.. also weirdly verify-vendir should fail on this change

**Comment by danwinship** [comment]
> /hold
> We are modifying the vendor here..

Yes, @smarterclayton had added some carry/drop patches in #2847; the upstream PRs (still not merged) are https://github.com/terraform-providers/terraform-provider-azurerm/pull/5139 and https://github.com/terraform-providers/terraform-provider-azurerm/pull/5590 and it looks like this touches both of them.

**Comment by danwinship** [comment]
fixes the crash

```
INFO Destroying the bootstrap resources...        
INFO Waiting up to 30m0s for the cluster at https://api.dwinship-ipv6-43.sdn.azure.devcluster.openshift.com:6443 to initialize... 
```

and I don't see any stray resources in the Azure portal that look like un-cleaned-up bootstrap resources.

File: pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_loadbalancer.go [hunk 1]
```diff
@@ -68,10 +68,9 @@ func resourceArmLoadBalancer() *schema.Resource {
 						},
 
 						"private_ip_address": {
-							Type:         schema.TypeString,
-							Optional:     true,
-							Computed:     true,
-							ValidateFunc: validate.IPv4AddressOrEmpty,
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
```
Comment (danwinship) [inline pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_loadbalancer.go]: I assume there should still be a `ValidateFunc`; you don't want it to accept random strings

File: pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_network.go [hunk 2]
```diff
@@ -416,8 +448,9 @@ func resourceAzureSubnetHash(v interface{}) int {
 
 	if m, ok := v.(map[string]interface{}); ok {
 		buf.WriteString(m["name"].(string))
-		buf.WriteString(m["address_prefix"].(string))
-
+		if v, ok := m["address_prefix"]; ok {
+			buf.WriteString(v.(string))
+		}
```
Comment (danwinship) [inline pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_network.go]: Not sure what this hash gets used for but you might need to use the `address_prefixes` value here, to ensure it stays unique

File: pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_loadbalancer.go (hunk 1 above)
Comment (jhixson74) [reply pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_loadbalancer.go]: I would have thought so as well, however, when poking through the terraform source, there is no validation for all the other private_ip_address fields scattered around the source.

File: pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_network.go (hunk 2 above)
Comment (jhixson74) [reply pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_network.go]: Apparently it is used to uniquely identify the resource. I've looked around at many of the other hashes and they leave out elements. However, I have updated it to use address_prefixes. I'm testing it and if all goes well I'll update the PR.

**Comment by jhixson74** [comment]
Moved here: https://github.com/openshift/terraform-provider-azurerm/pull/1

**Pull request closed without merging**